### DISCUSSION TITLE
Passing the list of flags to a script to handle parsing outside of Command class

### DISCUSSION THREAD

**Discussion by WillDelish** [oclif/oclif#1483] [Q&A] [answered]
Hello! This request might be the wrong way to solve this problem, but here is what I'm trying to do:

I have this parent command
```
export default class Device extends Command {

    static flags = {....}

    async run(): Promise<void> {
        const { flags } = await this.parse(Device)
```

So flags has all the correct types after its been parsed. All the sub commands re-use these flags and its easy to import the parent command flags and just parse them. The problem is that I'm creating a lot of repeated code handling all the flags parsing. I'm trying to create a separate `cmdHelper.ts` file that I could just pass the flags into and return the output. The problem is that before parsing the flags, they show the types as `OptionFlag<string | undefined, CustomOptions>;` vs just `token: string | undefined;`

I have a gut feeling there's an obvious way to go about this, I'm just missing it. Would love a recommendation!

**Comment by mdonnalley** [comment, marked as answer]
Take a look at the docs on [Custom Base Classes](https://oclif.io/docs/base_class)

**Reply by WillDelish** [reply]
Thanks!! `type QueryFlags = Interfaces.InferredFlags<typeof Device.flags>` is exactly what I wanted. I looked at this code and had a gut feeling something was. there, but not the confidence to really look.

Thanks for the quick reply also, I've really enjoyed using oclif. This library and the docs are very good!
